### PR TITLE
Strftime Jinja2 filter

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add a builtin Jinja2 filter for formatting ISO8601 datetime strings using strftime strings.

Some users are currently doing nasty string manipulation with the initial cycle point (as a Jinja2 variable).

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Worth having a test (validate and/or run) for using this filter in a suite.

File: lib/Jinja2Filters/strftime.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from isodatetime.parsers import (TimePointParser)  # Put in path by cylc.
```
Comment (matthewrmshin) [inline lib/Jinja2Filters/strftime.py]: Please add standard header.

Should not need to have brackets around `(TimePointParser)`?

**Comment by matthewrmshin** [comment]
A bit unrelated, but while we are at this, can you add `lib/Jinja2Filters/` to `tests/pep8/00-bin-lib.t`. There are some minor issues with `lib/Jinja2Filters/pad.py` apparently, but should be easy to fix.

**Comment by oliver-sanders** [comment]
* Added `lib/Jinja2Filters` to pep8 test.
* Made `pad.py` pep8 compliant.
* `tests/jinja2/09` now runs doctest on all Jinja2Filters.

**Comment by arjclark** [comment]
This looks really useful and should cut down on the various Jinja2 bits of code for manipulating datestrings that are going round the suites at the moment.

**Comment by matthewrmshin** [comment]
(Travis CI test is failing, but the failures are usual culprits, so can be ignored for now. I'll restart the test now. I'll investigate why these tests keep failing on Travis CI again after #2141.)

**Comment by oliver-sanders** [comment]
Added in-context tests for both filters.

**Comment by matthewrmshin** [comment]
Looks good.

File: doc/cug.tex [hunk 2]
```diff
@@ -4911,8 +4911,12 @@ \subsubsection{Custom Jinja2 Filters}
...
-needed. Currently there is one custom filter called ``pad'' in the
-central cylc Jinja2 filter directory, for padding string values to some
+needed. Currently there are two custom filters:
+
+
+\paragraph{pad}
+
+The ``pad'' filter is for padding string values to some
 constant length with a fill character - useful for generating task names
 and related values in ensemble suites:
```
Comment (hjoliver) [inline doc/cug.tex]: Incidentally, I have a feeling Jinja2 has a built in way of padding, so this is unnecessary. However, this isn't new so not an issue for this PR.

**Pull request merged**